common/src/main/java/com/box/l10n/mojito/openai/OpenAIClient.java

```diff
@@ -236,6 +251,61 @@ public interface Message {
       String name();
     }
 
+    public record ResponseFormat(String type) {
+      public static class ResponseFormatBuilder {
+        private String type;
+
+        private ResponseFormatBuilder() {}
+
+        public ResponseFormatBuilder type(String type) {
+          this.type = type;
+          return this;
+        }
+
+        public ResponseFormat build() {
+          return new ResponseFormat(type);
+        }
+      }
+
+      public static ResponseFormatBuilder responseFormatBuilder() {
+        return new ResponseFormatBuilder();
+      }
+    }
+
+    public record AssistantMessage(String role, String content, String name) implements Message {
```

Comment:
SystemMessageBuilder/AssistantMessageBuilder/UserMessageBuilder - all seem a bit duplicated.
Are of all them separate builders needed / could we use a single MessageBuilder that's shared - and based on the role returns the specific typed record required in build()?

Reply:
All of the different messages implement a Message interface, the only point of difference is that the role is set accordingly for each message type. I can remove the specific builder types and instead just have a single message builder which takes a role as an option (and defaults to user if none provided)